cmd/addvalidators.go

```diff
@@ -36,20 +54,208 @@ func newAddValidatorsCmd(runFunc func(context.Context, addValidatorsConfig) erro
-	return validateConf(conf)
+func runAddValidatorsSolo(conf addValidatorsConfig) (err error) {
+	// Read lock file to load mutable cluster state.
+	cState, err := loadClusterState(conf)
+	if err != nil {
+		return err
+	}
+
+	if err = validateConf(conf, cState.Operators); err != nil {
+		return err
+	}
+
+	// If a single address is provided, use the same address for all the validators.
+	if len(conf.FeeRecipientAddrs) == 1 {
+		var feeRecipients, withdrawalAddrs []string
+		for i := 0; i < conf.NumVals; i++ {
+			feeRecipients = append(feeRecipients, conf.FeeRecipientAddrs[0])
+			withdrawalAddrs = append(withdrawalAddrs, conf.WithdrawalAddrs[0])
+		}
+
+		conf.FeeRecipientAddrs = feeRecipients
+		conf.WithdrawalAddrs = withdrawalAddrs
+	}
+
+	// Generate new validators
+	var vals []state.Validator
+	for i := 0; i < conf.NumVals; i++ {
+		// Generate private/public keypair
+		secret, err := tbls.GenerateSecretKey()
+		if err != nil {
+			return errors.Wrap(err, "generate secret key")
+		}
+
+		pubkey, err := tbls.SecretToPublicKey(secret)
+		if err != nil {
+			return errors.Wrap(err, "generate public key")
+		}
+
+		// Split private key and generate public keyshares
+		shares, err := tbls.ThresholdSplit(secret, uint(len(cState.Operators)), uint(cState.Threshold))
+		if err != nil {
+			return errors.Wrap(err, "threshold split key")
+		}
+
+		var pubshares [][]byte
+		for _, share := range shares {
+			pubshare, err := tbls.SecretToPublicKey(share)
+			if err != nil {
+				return errors.Wrap(err, "generate public key")
+			}
+
+			pubshares = append(pubshares, pubshare[:])
+		}
+
+		feeRecipientAddr, err := eth2util.ChecksumAddress(conf.FeeRecipientAddrs[i])
+		if err != nil {
+			return errors.Wrap(err, "invalid fee recipient address")
+		}
+
+		withdrawalAddr, err := eth2util.ChecksumAddress(conf.WithdrawalAddrs[i])
+		if err != nil {
+			return errors.Wrap(err, "invalid withdrawal address")
+		}
+
+		// Generate builder registration
+		builderReg, err := builderRegistration(secret, pubkey, feeRecipientAddr, cState.ForkVersion)
+		if err != nil {
+			return err
+		}
+
+		vals = append(vals, state.Validator{
+			PubKey:              pubkey[:],
+			PubShares:           pubshares,
+			FeeRecipientAddress: feeRecipientAddr,
+			WithdrawalAddress:   withdrawalAddr,
+			BuilderRegistration: builderReg,
+		})
+	}
+
+	// Perform a `gen_validators/v0.0.1` mutation using the newly created validators.
+	genVals, err := state.NewGenValidators(cState.Hash, vals)
+	if err != nil {
+		return errors.Wrap(err, "generate validators")
+	}
+
+	genValsHash, err := genVals.Hash()
+	if err != nil {
+		return err
+	}
+
+	var (
+		enrKeys   []*k1.PrivateKey
+		approvals []state.SignedMutation
+	)
+	for _, enrKeyFile := range conf.EnrPrivKeyfiles {
+		enrKey, err := k1util.Load(enrKeyFile)
+		if err != nil {
+			return errors.Wrap(err, "load enr private key")
+		}
+
+		enrKeys = append(enrKeys, enrKey)
+	}
+
+	if conf.TestConfig.Lock != nil {
```

Comment:
doesn't load keys from disk if you provide them via test config

Reply:
Right, refactored the load keys functionality to a separate function